### ISSUE TITLE
Support focus style for focusable components

### ISSUE THREAD

**Issue by gokulkrishh** [tremorlabs/tremor-npm#305]
I am using DateRangepicker component in an app, it is focusable but there is no indicator for it like border or outline.

This is issue is request for supporting it, example i used here is DateRangepicker but ideally it should be supported by all focusable inputs, buttons etc,.

**How it is now**

![Screenshot 2023-02-16 at 9 05 01 AM](https://user-images.githubusercontent.com/2944237/219262026-774775f9-55c7-4d81-a796-d637c2d423cf.jpg)

**How it can be**

![Screenshot 2023-02-16 at 9 05 30 AM](https://user-images.githubusercontent.com/2944237/219262068-892f625a-6a0c-4833-921c-d98cfc92af06.jpg)

![Screenshot 2023-02-16 at 9 05 50 AM](https://user-images.githubusercontent.com/2944237/219262074-11c904d8-6c87-4e1b-9516-71c3a603f897.jpg)

**Comment by birjj** [comment]
This is a general accessibility concern that tremor doesn't seem to have handled. Just to add a bit to the list, the following components have problematic styling for keyboard navigation in either Chrome or Firefox:

_(sad cat 😿 indicates problem that makes keyboard interaction near unusable, usually due to removed focus indicator)_

<details><summary>Accordion 😿</summary>

Corners of outline cut off on Chrome, no outline in Firefox (😿):

![image](https://user-images.githubusercontent.com/4542461/219324365-974417f5-6c70-4b60-847b-b390f184d54f.png)

</details>
<details><summary>Buttons</summary>

Normal buttons have good focus styling (🎉), but light variants have a white outline that is almost invisible in both Chrome and Firefox:

![image](https://user-images.githubusercontent.com/4542461/219327342-2219f891-d839-43e7-994a-1f6669fa5b33.png)

</details>
<details><summary>Date Range Picker 😿</summary>

No outline for root elements in Chrome (😿). Good outline in Firefox (🎉):

![image](https://user-images.githubusercontent.com/4542461/219328279-652d2b52-2816-4543-a9ce-0a9f26ff35bd.png)

Good (but inconsistent) outlines for internal elements in both Chrome and Firefox.

Slightly nit-picky, but keyboard interaction in the date picker would be nice, so <kbd>Up</kbd> / <kbd>Down</kbd> / <kbd>Left</kbd> / <kbd>Right</kbd> works, would be nice. The current native implementation (<kbd>Tab</kbd> / <kbd>Enter</kbd> supported) works, but isn't necessarily intuitive.

</details>
<details><summary>Dropdown 😿</summary>

No outline in Chrome (😿). Good outline in Firefox (🎉):

![image](https://user-images.githubusercontent.com/4542461/219329182-707c5569-d7e1-4529-b859-88e9a4acccea.png)

The options in the dropdown have an outline shown, but they are cut off:

![image](https://user-images.githubusercontent.com/4542461/219331809-b852b811-2505-46c7-945e-28b676989a7b.png)

The dropdown has keyboard interaction implemented ***(🎉!!!)***, so <kbd>Up</kbd>, <kbd>Down</kbd> and <kbd>Esc</kbd> works - but the styling is different when using <kbd>Tab</kbd>.

</details>
<details><summary>Text input</summary>

No outline in Chrome or Firefox. There is a very weak focus indicator (blinking cursor), but that is insufficient:

![image](https://user-images.githubusercontent.com/4542461/219329560-751e808a-2ad1-45c3-9e0a-3cc5b31e8028.png)

</details>
<details><summary>Select 😿</summary>

**For single selects:**  
The currently chosen text is targetable in both Chrome and Firefox. It is usually used to filter the list of options when the select is open, but does nothing when the select is closed. It also does not open the select when hitting <kbd>Enter</kbd>.

The arrow is focusable and opens the select (🎉). The options in the dropdown have an outline shown, but they are cut off:

![image](https://user-images.githubusercontent.com/4542461/219331809-b852b811-2505-46c7-945e-28b676989a7b.png)

The select has keyboard interaction implemented ***(🎉!!!)***, so <kbd>Up</kbd>, <kbd>Down</kbd> and <kbd>Esc</kbd> works - but the styling is different when using <kbd>Tab</kbd>.

**For multi selects:**  
No outline in Chrome (😿):

![image](https://user-images.githubusercontent.com/4542461/219334432-4af519b6-e832-4b72-8b66-03c8954b0e22.png)

Internal outlines work in both Chrome and Firefox.

The select has keyboard interaction implemented ***(🎉!!!)***, but there is an inconsistency: if you use native interaction (<kbd>Tab</kbd>), then <kbd>Enter</kbd> doesn't select the option, <kbd>Space</kbd> does. If you use the custom implementation (<kbd>Up</kbd> / <kbd>Down</kbd>) then it's the inverse.

Slightly nit-picky, but it would be nice if the search field was focused when the multi-select is opened, so users can immediately type to search.

</details>
<details><summary>Tabs 😿</summary>

No outline in Chrome (😿):
![image](https://user-images.githubusercontent.com/4542461/219332661-a06516a2-a327-42a5-8fbc-de85b36b6e45.png)

</details>
<details><summary>Toggle (nit-pick)</summary>

Slightly nit-picky, but keyboard interaction would be nice, so <kbd>Left</kbd> / <kbd>Right</kbd> works, would be nice. The current native implementation (<kbd>Tab</kbd> / <kbd>Enter</kbd> supported) works, but isn't necessarily intuitive.

Outlines are shown in both Chrome and Firefox (🎉)

</details>

**Comment by gokulkrishh** [comment]
@birjj Thanks for sharing your thoughts and inputs. Having a accessible inputs should also be goal a of library imo. Lets wait for Tremor team to reply and see what they have in mind to told this problem.

**Comment by christopherkindl** [comment]
Thank you for raising this concern! Also congrats to @gokulkrishh for the launch of expense.fyi ☺️

Accessibility is not covered yet as we are still in beta. We are currently incorporating it for the production-ready version that will be released soon.

Thank you for patience and using our library! 🙏

**Comment by gokulkrishh** [comment]
@christopherkindl thanks and glad to know that A11Y support is in roadmap. Looking forward for the production version. Tremor is awesome keep up the good work. 🫡

**Comment by mitrotasios** [comment]
Yep, a lot of the points noted by @birjj will be addressed in the v2 which is supposed to be released at the end of this month. Thanks for listing these!

**Comment by severinlandolt** [comment]
WIP:

Focus Styling Status: 🟢🟠🔴

- Accordion: Chrome🟠 Safari🔴 Firefox🔴
- Buttons: Chrome🔴 Safari🔴 Firefox🔴
- Date (Range) Picker Chrome🟢 Safari🟢 Firefox🟢 (improvements to date range selection)
- Text input: Chrome🟢 Safari🟢 Firefox🟢
- Select TODO
    - Select
    - SearchSelect
    - MultiSelect
- Tabs TODO
- Number Input: Chrome🟢 Safari🟢 Firefox🟢
- ~~Toggle (nit-pick)~~ deprecated

**Comment by adnjoo** [comment]
+1 for tab

I saw the base is [headlessui](https://headlessui.com/react/tabs) tab so you can use e.g. ⬅️➡️ arrow keys to navigate through the tabs
